### PR TITLE
Access token automatic generation

### LINKED ISSUE #689: Automatic generation of new access_token
As specd [here](https://github.com/stormpath/stormpath-framework-spec/blob/master/cookie-authentication.md#authentication-flow), if an access token is invalid attempt to use the refresh token to get a new access token.

### PR DESCRIPTION
**Pull request by mrioan**
Resolves #689 

### Pre-step
```
http -f POST http://localhost:8080/oauth/token Origin:http://localhost:8080 grant_type=password username=YOUR_EMAIL password=YOUR_PASS
```
Take note of access and refresh token obtained

### Test 1:
```
http localhost:8080 Cookie:"access_token=$ACCESS_TOKEN;refresh_token=$REFRESH_TOKEN"
```
Expected outcome: Should show the home page with the name

### Test 2:
```
http localhost:8080 Cookie:"refresh_token=$REFRESH_TOKEN"
```
Expected outcome: Should show the home page with the name and a new access and refresh token should be received

### Test 3
```
http localhost:8080 Cookie:"access_token=FOO_TEXT_HERE;refresh_token=$REFRESH_TOKEN"
```
Expected outcome: Should show the home page with the name and a new access and refresh token should be received

### Test 4
```
http localhost:8080 Cookie:"access_token=$ACCESS_TOKEN"
```
Expected outcome: Should show the home page with the name

### Test 5
```
http localhost:8080 Cookie:"refresh_token=$ACCESS_TOKEN"
```
Expected outcome: Should show the home page with "Hello World"

### Test 6
```
http localhost:8080 Cookie:"access_token=$REFRESH_TOKEN"
```
Expected outcome: Should show the home page with "Hello World" (no account identified)

### Test 7
```
http localhost:8080
```
Expected outcome: Should show the home page with "Hello World" (no account identified)

These steps should be carried out with `spring-security-webmvc`, `servlet` and `spring-boot-webmvc`

### CODE REVIEW

**Comment by mzumbadoate** [comment]
UAT's are working ok
QUESTION:
once I use a refresh token to generate new tokens, should "old tokens" still work, my guess is yes if the expiration date is still valid.
This means accumulating tokens as long as they don't expire this sounds ok but I'm asking.

**Comment by mzumbadoate** [comment]
LGTM

**Comment by lhazlewood** [comment]
Yes, old tokens that haven't expired and haven't been revoked can still be used.  HTH!

File: extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/account/CookieAccountResolver.java [hunk 1]
```diff
@@ -87,10 +108,46 @@ protected Account getAccount(HttpServletRequest request, HttpServletResponse res
...
+        String val = cookie.getValue();
+        if (!Strings.hasText(val)) {
+            return null;
+        }
+
+        try {
+            OAuthRefreshTokenRequestAuthentication refreshGrantRequest = OAuthRequests.OAUTH_REFRESH_TOKEN_REQUEST.builder()
+                    .setRefreshToken(val)
+                    .build();
+            OAuthRequests.OAUTH_REFRESH_TOKEN_REQUEST.builder().setRefreshToken(refreshTokenCookieResolver.get(request, response).getValue());
```
Comment (dogeared) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/account/CookieAccountResolver.java]: @mrioan - This PR is already closed, but line `132` seems redundant. I think it should be deleted. Do you concur?

Comment (mrioan) [inline extensions/servlet/src/main/java/com/stormpath/sdk/servlet/filter/account/CookieAccountResolver.java]: @dogeared good catch. Yes, duplicated code there. Thanks for the heads up. I'll improve this in any future commit...

**Pull request merged**